crates/utils/src/settings/structs.rs

```diff
@@ -101,8 +101,9 @@ pub struct EmailConfig {
   #[doku(example = "noreply@example.com")]
   /// Address to send emails from, eg "noreply@your-instance.com"
   pub smtp_from_address: String,
-  /// Whether or not smtp connections should use tls
-  pub use_tls: bool,
+  /// Whether or not smtp connections should use tls. Can be missing (IE None), tls, or starttls
+  #[default(None)]
+  pub use_tls: Option<String>,
```

Comment:
Some possibilities here: 

This could just be made a mandatory string, maybe `tls_type`, and be `none`, `starttls`, or `tls`. That might be simpler than an option string.

Reply:
It should simply be an enum. Also you need to add the `#doku(example)`.

Reply:
There are no other enums here, and I doubt doku would be able to convert from a string to that. the `difficulty` field is probably the closest thing here, and that also doesn't have an example for some reason, just a comment.

Reply:
Doku just works with strings, so the Rust type doesnt matter. And difficulty should also be an enum with comment to be honest.

Reply:
I have no idea how enums would work reading them from an hjson config file. But I can change this to be a mandatory string rather than an option for clarity.

Reply:
They just need to derive serde::Serialize/Deserialize (and probably add an annotation to make it lowercase).